src/main/java/beam/analysis/plots/RealizedModeAnalysis.java

```diff
@@ -516,6 +521,26 @@ public void writeToReferenceCSV() {
         }
     }
 
+    public void writeReplanningReasonCountCSV(Integer iteration) {
+        String fileName = GraphsStatsAgentSimEventsListener.CONTROLLER_IO.getIterationFilename(iteration,"replanningEventReason.csv");
+        String header = "ReplanningReason,Count";
+        try{
+            FileWriter writer = new FileWriter(new File(fileName));
+            BufferedWriter out = new BufferedWriter(writer);
+            out.write(header);
+            out.newLine();
+            for (Map.Entry<String,Integer> entry : replanningReasonCount.entrySet()){
+                out.write(entry.getKey()+","+entry.getValue());
+                out.newLine();
+            }
+            out.close();
```

Comment:
These should use the disposal pattern for safety

````
try(FileWriter writer = new FileWriter(new File(fileName));
  BufferedWriter out = new BufferedWriter(writer)) {
...
} catch ...
````

Reply:
this is done